timesketch/frontend/src/components/SketchExploreSessionChart.vue

```diff
@@ -0,0 +1,308 @@
+    },
+
+    selectSessionID: function (event) {
+        var queryString = ''
+        var selection = []
+
+        var selectedSessionsCopy = JSON.parse(JSON.stringify(this.selectedSessions))
+        for (var i = 0; i < selectedSessionsCopy.length; i++) {
+            var session = selectedSessionsCopy[i]
+            if (this.selectedType === session.session_type && this.selectedID === session.session_id) {
+                session['selected'] = true
+                queryString = session['session_type'] + ':' + session['session_id']
+                selection = [session['start_timestamp'] - (this.barSize * 5), session['end_timestamp'] + (this.barSize * 5)]
+            }
+            else {
+                session['selected'] = false
+            }
+        }
+
+        var dictSpec = JSON.parse(this.spec)
+        dictSpec['data']['values'] = selectedSessionsCopy
+        dictSpec['vconcat'][1]['selection']['brush']['init']['x'] = selection
+        this.spec = JSON.stringify(dictSpec)
+
+        this.$store.commit('updateCurrentQueryString', queryString)
+        this.$store.commit('search', this.sketch.id)
+    },
+
+    getProcessedSessions: function (first_timestamp, last_timestamp) {
+        //increases the visibility of sessions when plotted
+        var processedSessions = JSON.parse(JSON.stringify(this.sessions))
+        this.isTruncated = processedSessions.pop()['truncated']
+        
+        processedSessions = processedSessions.filter(session => session.start_timestamp >= first_timestamp && session.start_timestamp <= last_timestamp);
+
+        if (processedSessions.length > 0) {
+            var smallestTimestamp = processedSessions[0].start_timestamp
+            var largest_timestamp = processedSessions[0].end_timestamp
+            for (var i = 1; i < processedSessions.length; i++) {
+                if (processedSessions[i].start_timestamp < smallestTimestamp) {
+                    smallestTimestamp = processedSessions[i].start_timestamp
+                }
+                if (processedSessions[i].end_timestamp > largest_timestamp) {
+                    largest_timestamp = processedSessions[i].end_timestamp
+                }
+            }
+            var timeRange = (largest_timestamp - smallestTimestamp)
+
+            //session bars will take up 1/1000th of the chart if they would otherwise be smaller
+            this.barSize = Math.round(timeRange / 1000)
+            this.smallestTimestamp = smallestTimestamp
+
+            for (var i = 0; i < processedSessions.length; i++) {
+                var session = processedSessions[i]
+                var extended_end = session.start_timestamp + this.barSize
+                if (session.end_timestamp < extended_end) {
+                    session.end_timestamp = extended_end
+                }
+            }
+        }
+        this.processedSessions = processedSessions
+    },
+
+    getVegaSpec: function () {
+        const WIDTH = 900
```

Comment:
Can we make this responsive? Something like:
```
        spec.config.view.width = this.$refs.vegaChart.$el.offsetWidth
        spec.config.autosize = { type: 'fit', contains: 'padding' }
```

Reply:
Totally fine to just add a #TODO here and we can do that at a later stage.

Reply:
Added a TODO